uber/models.py

```diff
@@ -1343,6 +1354,36 @@ def full_name(cls):
         ], else_=func.lower(cls.first_name + ' ' + cls.last_name))
 
     @hybrid_property
+    def legal_name(self):
+        return self.unassigned_name or self.legal_name_ or '{self.first_name} {self.last_name}'.format(self=self)
+
+    @legal_name.expression
+    def legal_name(cls):
+        return case([
+            (or_(cls.first_name == None, cls.first_name == ''), 'zzz'),
+            (cls.legal_name_ != None, cls.legal_name_)
+        ], else_=func.lower(cls.first_name + ' ' + cls.last_name))
+
+    @legal_name.setter
+    def legal_name(cls, value):
+        # make sure value is not the empty string
+        if not value:
+            value = None
+        cls.legal_name_ = value
+
+    # This property should be used in situations where we specifically want an empty string when legal_name is unset
+    # (as opposed to falling back to full_name
```

Comment:
See my above comment about disallowed NULL and defaulting to the empty string when this isn't set.  This kind of property is problematic because every future programmer who interacts with this field will need to know and remember the difference between ``attendee.legal_name`` and ``attendee.legal_name_opt``, which will not be intuitive and will be easy to forget.  Whereas if we simply default to the empty string then all of our checks for whether ``attendee.legal_name`` is falsey will still work (e.g. ``attendee.legal_name or attendee.full_name``) but we won't have created the possibility to use ``None`` where we meant to use ``''``.